### ISSUE TITLE
Timepoint and Duration are interpreted as user defined types instead of SDK types in the WASM spec XDR

### ISSUE THREAD

**Issue by wildework** [stellar/rs-soroban-sdk#1196]
### What version are you using?

```toml
soroban-sdk = "20.0.0"
```

CLI

```
soroban 20.1.1 (deb1f7cbdc4113dbedffe981fad0dff3d5945d5c)
soroban-env 20.0.2 (81f6eb1eefd299af3e03fa0db7b20eb355b2b55f)
soroban-env interface version 85899345920
stellar-xdr 20.0.2 (a928e82943e4e93c887555fdec3b1e139ad810d2)
xdr curr (bb54e505f814386a3f45172e0b7e95b7badbe969)
```

### What did you do?

A basic Soroban contract that takes and returns `Timepoint` and `Duration` values.

```rust
#![no_std]
use soroban_sdk::{contract, contractimpl, Env, Timepoint, Duration};

#[contract]
pub struct Contract;

#[contractimpl]
impl Contract {
    pub fn read_timepoint(env: Env, time: Timepoint) -> Timepoint {
        time
    }
    pub fn read_duration(env: Env, duration: Duration) -> Duration {
        duration
    }
}
```

### What did you expect to see?

I expected to see a spec generated where types were `scSpecTypeTimepoint` and `scSpecTypeDuration`.

### What did you see instead?

These are the bindings I got after running `soroban contract bindings json --wasm`. But I do get the same results even when parsing the raw XDR straight from the WASM generated by `cargo build`, so it's definitely not CLI related.

```json
[
  {
    "type": "function",
    "doc": "",
    "name": "read_timepoint",
    "inputs": [
      {
        "doc": "",
        "name": "time",
        "value": {
          "type": "custom",
          "name": "Timepoint"
        }
      }
    ],
    "outputs": [
      {
        "type": "custom",
        "name": "Timepoint"
      }
    ]
  },
  {
    "type": "function",
    "doc": "",
    "name": "read_duration",
    "inputs": [
      {
        "doc": "",
        "name": "duration",
        "value": {
          "type": "custom",
          "name": "Duration"
        }
      }
    ],
    "outputs": [
      {
        "type": "custom",
        "name": "Duration"
      }
    ]
  }
]
```

**Comment by leighmcculloch** [comment]
> I do get the same results even when parsing the raw XDR straight from the WASM generated by cargo build, so it's definitely not CLI related.

You're correct this isn't an issue with the CLI. It's an issue with the Rust soroban-sdk crate and the contract interface/spec it has generated.

It is generating incorrect contract interface/spec XDR for Timepoint types. The SDK contains a map of type names to ScSpecTypeDef's and we're missing a branch for Timepoint and Duration that capture them as SDK/host types instead of as custom user defined types.

The CLI and the Rust soroban-sdk crate should both generate the correct code for Timepoint/Duration if the XDR is correct. At least according to the code as I see it.

The code I tested with was:

```rust
#![no_std]
use soroban_sdk::{contract, contractimpl, Timepoint};

#[contract]
pub struct Contract;

#[contractimpl]
impl Contract {
    pub fn empty(_t: Timepoint) {}
}
```

And this is the resulting contract interface/spec XDR:

```
$ soroban contract inspect --wasm target/wasm32-unknown-unknown/release/contract.wasm
Env Meta: AAAAAAAAABQAAAAA
 • Interface Version: 85899345920

Contract Meta:
 • rsver: 1.75.0
 • rssdkver: 20.0.3#93b09e42e4efa841cbd034c0bff0dc362765086c-dirty

Contract Spec:
 • Function: empty
     Inputs: VecM(
         [
             ScSpecFunctionInputV0 {
                 doc: StringM(),
                 name: StringM(_t),
                 type_: Udt(
                     ScSpecTypeUdt {
                         name: StringM(Timepoint),
                     },
                 ),
             },
         ],
     )
     Output: VecM(
         [],
     )
```

The `type_` line should be a value indicating Timepoint instead of a user-defined-type (UDT) with name "Timepoint".

Independent of this bug, it also appears that the SDK doesn't make it easy to construct a Timepoint or Duration. I opened a separate issue for that:
- https://github.com/stellar/rs-soroban-sdk/issues/1197

These types have not had much use it would seem as these issues have existed for sometime.

**Comment by leighmcculloch** [comment]
Will be fixed by:
- https://github.com/stellar/rs-soroban-sdk/pull/1200

**Comment by leighmcculloch** [comment]
Fix released in [v20.2.0](https://github.com/stellar/rs-soroban-sdk/releases/tag/v20.2.0).
